File: Chavez-Rubio-Javier/P2/Python/fork.py
#	$ python3 fork.py

import os

pid = os.fork()

if pid == -1:
    print("Error de creación")
elif pid == 0:
    print(f"PID del hijo: {os.getpid()}")
else:
    print(f"PID del padre: {os.getpid()}")

File: Chavez-Rubio-Javier/P1/Cpp/fork.cpp
/*
    $ g++ -o fork fork.cpp
    $ ./fork
*/

#include <iostream>
#include <unistd.h>

int main() {
    pid_t pid;
    pid = fork();
    
    switch (pid) {
        case -1:
            std::cerr << "Error de creación" << std::endl;
            break;
        case 0:
            std::cout << "Código del hijo: " << getpid() << std::endl;
            break;
        default:
            std::cout << "Código del padre: " << getpid() << std::endl;
            break;
    }
    
    return 0;
}
